### ISSUE TITLE
single-component-path-imports should be disabled by default

### ISSUE THREAD

**Issue by ijackson** [rust-lang/rust-clippy#13420]
### Summary

#7168 #7290 #9382 #9718 #11925 #13419 

This lint is IMO too buggy and should not be enabled by default.

### Reproducer

I tried this code:

```rust
<code>
```

I expected to see this happen:

Instead, this happened:
